docs/manual/src/internals/ffi_converter_traits.md

```diff
@@ -0,0 +1,68 @@
+## Rust FFI conversion traits
+
+UniFFI leverages a set of FFI converter traits to implement [lifting and lowering](./lifting_and_lowering.md) on the Rust side.
+Each trait handles a single step in the lifting/lowering process (e.g. lifting an argument, lowering a return, etc.).
+We implement the trait for each type used in the exported API then use that trait in the codegen.
+
+For example, `uniffi::Lift` is used to lift values.
+To handle a function like `fn print(msg: String)`, the generated code will use:
+
+- `<String as Lift<crate::UniFfiTag>>::FfiType` when it needs to specify the FFI type (`RustBuffer`).
+- `<String as Lift<crate::UniFfiTag>>::try_lift()` when it needs to lift an argument value.
+
+Using a trait for this is important for proc-macros, which only see Rust tokens and don't know the surrounding context.
+For example, if macros always used `RustBuffer` as the FFI type whenever it sees "String", then that would fail if users created a type alias like `type String = u8`.
+This may be unusual for `String`, but it's very common for `Result`.
+In general, any reasoning about the tokens is fragile and should be avoided.
+
+### `UniFfiTag` and the orphan rule
+
+One odd part about the above code is that the `Lift` trait has a generic parameter which is always set to `crate::UniFfiTag`.
+In general, all of the FFI converter traits have this parameter (i.e. we generate `Lower<crate::UniFfiTag>`, `LowerReturn<crate::UniFfiTag>`, etc.).
+What's the point of all of this?
+
+The main reason is to work around issues with the Rust orphan rule and types from 3rd-party crates.
+For example, the [custom types](../types/custom_types.md) documentation shows how `url::Url` can be used in an exported API.
+For these types, we normally can't implement `Lift` in the generated code since neither `uniffi::Lift` or `url::Url` is local to the crate.
```

Comment:
s/generated code/code we generate in the crate/ here, to help keep the focus on that orphan rule?